src/rgw/rgw_lua.cc

```diff
@@ -0,0 +1,135 @@
+#include <lua.hpp>
+#include "rgw_lua.h"
+#include "rgw_common.h"
+#include "common/dout.h"
+#include <iostream>
+#include <mutex>
+
+#define dout_subsys ceph_subsys_rgw
+
+namespace rgw::lua {
+
+class lua_state_guard {
+  lua_State* l;
+public:
+  lua_state_guard(lua_State* _l) : l(_l) {}
+  ~lua_state_guard() {lua_close(l);}
+  void reset(lua_State* _l=nullptr) {l = _l;}
+};
+
+int RGWLog(lua_State* L) {
+  auto cct = (CephContext*)lua_touserdata(L, lua_upvalueindex(1));
```

Comment:
static_cast (usuall) or reinterpret_cast ?

Reply:
its a void*, will change to reinterpret_cast